### ISSUE TITLE
"Discard current input?" is not suitable for existing-content-editing screens ("Edit note", "Deck options", "Card type")

### ISSUE THREAD

**Issue by snowtimeglass** [ankidroid/Anki-Android#15788]
> I would also propose to change the wording to "Discard changes?" in case changes were made, since any user actions in the deck options are rather "changes" and not "input". Should I open an extra issue for that?

_Originally posted by @camoz in https://github.com/ankidroid/Anki-Android/issues/15782#issue-2169625911_

> What about changing "Discard current input?" to "Discard changes?" or similar (cf. last paragraph I wrote)?

_Originally posted by @camoz in https://github.com/ankidroid/Anki-Android/issues/15782#issuecomment-1979260121_

The opinion seems right not only from the viewpoint of meaning but also from the viewpoint of consistency with Anki Desktop.

Actually, in Anki Desktop, "Discard current input?", is used just for "Add" screen; it isn't used for the following screens: "Edit note", "Deck options", "Card type". 

---

![image](https://github.com/ankidroid/Anki-Android/assets/10436072/d85ca27e-54a6-4346-96ff-43634e93d407)

---

![image](https://github.com/ankidroid/Anki-Android/assets/10436072/637c5b78-d088-4418-94d6-db04900cb529)

https://github.com/search?q=repo%3Aankitects%2Fanki+%22adding_discard_current_input%22&type=code

---

It means that this string is used just in case that a new note is created and is input something, not in cases that existing (= already-saved) contents is edited. I'd say that this string is optimized for new creation and is not suitable for other situations.

However, in AnkiDroid, "Discard current input?" is used not only for "Add" screen, but also for those existing-content-editing screens ("Edit note", "Deck options", "Card type"). In these cases, this message looks a bit odd and less intuitive, as @camoz may feel. 

So, for the screens ("Edit note", "Deck options", "Card type"), it will probably be better to reuse "Discard _changes_?" from Anki Desktop. It will work more clearer, more intuitive, less confusing in the situations, as it works in Anki Desktop as follows:

![image](https://github.com/ankidroid/Anki-Android/assets/10436072/adc91cb5-a628-463d-89aa-98ceebf9974d)

**Comment by user1823** [comment]
I agree that it's better to reuse "Discard changes?" from Anki Desktop in cases where **existing** content is being edited.

However, the options "Discard" and "Keep Editing" should remain unchanged.

**Comment by camoz** [comment]
I think this would still be nice to have.

Can we reopen this?

**Comment by user1823** [comment]
Still looking for a solution...

**Comment by camoz** [comment]
Still looking for a solution...

**Comment by camoz** [comment]
Still looking for a solution...

**Comment by Alex7D3** [comment]
Hey @david-allison ! Looks like this is still open. Could I give this a go? It would be my first contribution.

**Comment by david-allison** [comment]
Sure, cheers!
